### PR TITLE
Mech updates part 3: grabbing furniture and wielding items

### PR DESCRIPTION
**Pull request by chaosvolt**
#### Summary

SUMMARY: Features "Unarmed mechs can now move furntiure, allow wielding weapons to fire from, supports heavy weapons"

#### Purpose of change

And this PR separates some additional expansions to mech functionality into its own PR to avoid bogging down the other two made recently, focusing on some things specifically pertaining to unarmed mechs like the lifter mech.

#### Describe the solution

1. In handle_action.cpp, changed `game::handle_action` to allow grabbing furniture/vehicles while piloting a mech if mech weapon is undefined. This allows the lifting mech to actually help with moving heavy things like its purpose indicates. You can shove vehicles in a mech it turns out, but obviously this change enables a more delicate approach. The code even has a function for adding your mech's strength bonus to moving furniture, but it went unused because of this restriction, derp.
2. In game.cpp, updated `game::grabbed_furn_move` to account for mech strength when moving furniture and vehicles. `map::can_move_furniture` already does this but turns out it's basically solely for determining if the grab message will claim the object feels heavy.
3. In grab.cpp, updated `game::grabbed_veh_move` in the same way as above, with mech strength affecting your ability to move vehicles.
3. In character.cpp, changed `Character::can_wield` so that you can opt to wield weapons if riding a mech that lacks a mech weapon. You'll still have to bring your own weapon, and the mech's melee attack will still take precedence over yours, but the idea is this allows for consistency with the ability to throw items in inventory, and makes it possible to fire a ranged weapon from within an unarmed mech.
4. In monexamine.cpp, set it so that `monexamine::pet_menu` allows you to enter unarmed mechs while wielding weapons, still barring you from doing so if the mech has its own weapon.
5. In ranged.cpp, set it so that `ranged::gunmode_checks_weapon` checks if you're mounted on a mech, and if so allows it as a stable platform for `MOUNTED_GUN` weapons, while still excluding normal mounts.
6. Updated the monster description of the lifter mech to reference the fact that it's unarmed but allows the use of ranged weapons while piloting it, and references the support it offers for `MOUNTED_GUN` weapons.

#### Describe alternatives you've considered

1. Allowing regular mounts to count as a valid platform for `MOUNTED_GUN` weapons, presumably at Large-tier penalty.
2. Making it so that firing a `MOUNTED_GUN` weapon is done at Large or Huge tier penalty instead of like mountable terrain.

#### Testing

1. Compiled and load-tested.
2. Tested grabbing furniture and vehicles in a lifter mech, especially confirming that it doesn't try to shove the latter and instead pushes them for you.
3. Tested that a dumpster that was too heavy to push for a standard character could be pushed while piloting a lifter mech.
4. Tested that dismounting correctly unsets your grab like the code suggests it should.
5. Tested that the recon mech still can't attempt to grab and that it says it lacks hands to do so.
6. Tested to confirm that riding a horse still precludes grabbing things full-stop.
7. Tested that you can board a lifter mech while wielding a weapon, and that neither boarding nor dismounting does anything weird to your wielded weapon.
8. Tested that you can switch weapons freely while in your mech, but can't use it to work around the inability to pickup items off the floor.
9. Tested that you can fire a ranged or reach weapon while in your mech.
10. Tested that you can freely fire an M2 browning while in a lifter mech, but not while on a horse.
10. Tested that you still need to unwield your weapon to mount a recon mech.
11. Tested that you're still bound from changing weapons in a recon mech by the item in question having NO_UNWIELD, and that it still correctly sets and unsets when mounting/dismounting said mech.
12. Tested that riding a horse isn't affected by and doesn't affect wielding weapons.
13. Checked affected files for astyle.

One issue encountered during testing: It won't let you pull grabbed object since the mech is considered to be blocking the way, only push.

### CODE REVIEW

File: src/handle_action.cpp [hunk 1]
```diff
@@ -1927,7 +1927,16 @@ bool game::handle_action()
                 if( u.has_active_mutation( trait_SHELL2 ) ) {
                     add_msg( m_info, _( "You can't grab things while you're in your shell." ) );
                 } else if( u.is_mounted() ) {
-                    add_msg( m_info, _( "You can't grab things while you're riding." ) );
+                    auto mon = u.mounted_creature.get();
+                    if( !mon->has_flag( MF_RIDEABLE_MECH ) ) {
+                        add_msg( m_info, _( "You can't grab things while you're riding." ) );
+                        break;
+                    } else if( !mon->type->mech_weapon.is_empty() ) {
```
Comment (Night-Pryanik) [inline src/handle_action.cpp:1934]: This check will make lifter mech (which is AFAIK designed to haul items) unable to grab things/furniture, as this mech doesn't have `mech_weapon` defined?

Comment (chaosvolt) [reply src/handle_action.cpp:1934]: `!` plus `is_empty` means it's checking if mech weapon is not empty (i.e. if it's armed). Double negative, basically.

Comment (Night-Pryanik) [reply src/handle_action.cpp:1934]: Oh, didn't notice the `!`.

**Comment by Coolthulhu** [comment]
Doesn't compile, looks like you forgot to push a commit with a fix or something.

**Comment by chaosvolt** [comment]
Hecc, could've sworn I did, will check when I get back later today.

**Comment by chaosvolt** [comment]
Okay, fixed the error I spotted and it once again compiles for me, so hopefully that was all it needed.

**Review by Firestorm01X2 — APPROVED** [review]
Ready

**Pull request merged**